### PR TITLE
THREESCALE-10296 - add OAuth2/OIDC to supported sec schemes in OpenAPI limi…

### PR DESCRIPTION
**Pull request by valerymo**
## WHAT
Jira: https://issues.redhat.com/browse/THREESCALE-10296

Doc updated - OIDC notes added to limitation section,  as requested here - https://github.com/3scale/3scale-operator/issues/886

### CODE REVIEW

File: doc/openapi-user-guide.md [hunk 1]
```diff
@@ -194,7 +194,14 @@ spec:
   * Only first `servers[0].url` element in `servers` list parsed as *private base url*. As OpenAPI specification `basePath` property, `servers[0].url` URL's base path component will be used.
   * `servers` element in path item or operation items are not supported.
   * Just a single top level security requirement supported. Operation level security requirements not supported.
-  * Supported security schemes: `apiKey`.
+  * Supported security schemes: `apiKey`. 
+* As detailed in section [OpenID Connect and OAuth2 example](#openid-connect-and-oauth2-example)  - 
+3scale requires some additional information that are not included in the OpenAPI spec, so need to be provided in the OpenAPI CR:
```
Comment (MStokluska) [inline doc/openapi-user-guide.md:199]:
```suggestion
3scale requires some additional pieces of information that are not included in the OpenAPI spec and need to be provided in the OpenAPI CR:
```

File: doc/openapi-user-guide.md [hunk 2]
```diff
@@ -194,7 +194,14 @@ spec:
...
+  * OpenID Connect Issuer Type, which will default to rest but can be overridden from the OpenAPI CR 
+  * OpenID Connect Issuer Endpoint Referense (Secret). 3scale requires that the issuer URL must include a client secret.
```
Comment (MStokluska) [inline doc/openapi-user-guide.md:201]:
```suggestion
  * OpenID Connect Issuer Endpoint Reference (Secret). 3scale requires that the issuer URL includes a client secret. 
```

File: doc/openapi-user-guide.md [hunk 3]
```diff
@@ -194,7 +194,14 @@ spec:
   * Only first `servers[0].url` element in `servers` list parsed as *private base url*. As OpenAPI specification `basePath` property, `servers[0].url` URL's base path component will be used.
   * `servers` element in path item or operation items are not supported.
   * Just a single top level security requirement supported. Operation level security requirements not supported.
-  * Supported security schemes: `apiKey`.
+  * Supported security schemes: `apiKey`. 
+* As detailed in section [OpenID Connect and OAuth2 example](#openid-connect-and-oauth2-example)  - 
+3scale requires some additional information that are not included in the OpenAPI spec, so need to be provided in the OpenAPI CR:
+  * OpenID Connect Issuer Type, which will default to rest but can be overridden from the OpenAPI CR 
+  * OpenID Connect Issuer Endpoint Referense (Secret). 3scale requires that the issuer URL must include a client secret. 
+  * Flows object. When the sec scheme is oauth2, the flows are provided by the OpenAPI doc. However, for openIdConnect security scheme, the OpenAPI doc does not provide the flows. In that case, the OpenAPI CR can provide those.
```
Comment (MStokluska) [inline doc/openapi-user-guide.md:202]: is `sec` security? 
if OpenAPI does not provide the flows, does OpenAPI must provide them or is it optional? If it must, change "can" to "must" please. If not it's fine

**Comment by MStokluska** [comment]
e2e fails due to https://issues.redhat.com/browse/THREESCALE-10277

File: doc/openapi-user-guide.md [hunk 4]
```diff
@@ -194,7 +194,14 @@ spec:
   * Only first `servers[0].url` element in `servers` list parsed as *private base url*. As OpenAPI specification `basePath` property, `servers[0].url` URL's base path component will be used.
   * `servers` element in path item or operation items are not supported.
   * Just a single top level security requirement supported. Operation level security requirements not supported.
-  * Supported security schemes: `apiKey`.
+  * Supported security schemes: `apiKey`.
```
Comment (mkudlej) [inline doc/openapi-user-guide.md:197]: missing  `oauth2` and/or `openIdConnect` in the list of supported security schemes

File: doc/openapi-user-guide.md [hunk 5]
```diff
@@ -194,7 +194,14 @@ spec:
   * Only first `servers[0].url` element in `servers` list parsed as *private base url*. As OpenAPI specification `basePath` property, `servers[0].url` URL's base path component will be used.
   * `servers` element in path item or operation items are not supported.
   * Just a single top level security requirement supported. Operation level security requirements not supported.
-  * Supported security schemes: `apiKey`.
+  * Supported security schemes: `apiKey`. 
+* As detailed in section [OpenID Connect and OAuth2 example](#openid-connect-and-oauth2-example)  -
```
Comment (mkudlej) [inline doc/openapi-user-guide.md:198]: add text below to a separated sub-section with title. Something like `openIdConnect/oauth2 limitations`

**Review by mkudlej — CHANGES_REQUESTED** [review]

File: doc/openapi-user-guide.md (hunk 1 above)
Comment (valerymo) [reply doc/openapi-user-guide.md]: updated. Thank you

File: doc/openapi-user-guide.md (hunk 2 above)
Comment (valerymo) [reply doc/openapi-user-guide.md]: updated. Thank you

File: doc/openapi-user-guide.md (hunk 3 above)
Comment (valerymo) [reply doc/openapi-user-guide.md]: It's optional. The default is false. As noted in Design: https://github.com/3scale/3scale-operator/issues/842 , where also example for OpenAPI CRD.  In code - it's here: https://github.com/3scale/3scale-operator/blob/master/controllers/capabilities/openapi_product_reconciler.go#L399-L403
Thank you

File: doc/openapi-user-guide.md (hunk 4 above)
Comment (valerymo) [reply doc/openapi-user-guide.md]: done. Thank you

File: doc/openapi-user-guide.md (hunk 5 above)
Comment (valerymo) [reply doc/openapi-user-guide.md:203]: done. Thank you

**Pull request merged**